src/id3/v2/tag.rs

```diff
@@ -708,19 +708,53 @@ impl SplitAndMergeTag for ID3v2Tag {
 
 		self.frames.reserve(tag.item_count() as usize);
 
-		if let Some(artists) = join_items(&mut tag, &ItemKey::TrackArtist) {
-			self.set_artist(artists);
-		} else {
-			self.remove_artist()
+		// TODO: Extend list of supported multi-valued text frames
+		for item_key in &[
+			&ItemKey::TrackArtist,
+			&ItemKey::AlbumArtist,
+			&ItemKey::TrackTitle,
+			&ItemKey::AlbumTitle,
+			&ItemKey::ContentGroup,
+			&ItemKey::AppleId3v2ContentGroup,
+			&ItemKey::Genre,
+			&ItemKey::Mood,
+			&ItemKey::Composer,
+			&ItemKey::Conductor,
+		] {
+			let frame_id = item_key
+				.map_key(TagType::ID3v2, false)
+				.expect("valid frame id");
+			if let Some(text) = join_text_items(&mut tag, item_key) {
+				let frame = Frame {
+					id: FrameID::Valid(Cow::Borrowed(frame_id)),
+					value: FrameValue::Text {
+						encoding: TextEncoding::UTF8,
+						value: text,
+					},
+					flags: FrameFlags::default(),
+				};
+				self.insert(frame);
+			} else {
+				self.remove(frame_id);
+			}
 		}
 
+		if let Some(text) = join_text_items(&mut tag, &ItemKey::Comment) {
+			// The first comment frame is either replaced or added.
+			debug_assert!(self.comments().count() <= 1);
```

Comment:
This assertion isn't always valid, since you can merge a Tag with an ID3v2Tag that already has comments inside of it.

Also, you should just take the first comment instead of concatenating all of them into one. I feel like the former is less surprising.

Reply:
The merge operation should only be invoked on the untouched, remaining part after splitting a tag. The remaining part does not contain any of those item keys, otherwise the results are undefined (as documented).

This is not supposed to be a universal merge operation. Otherwise the could would need to be less efficient and more complex.

Reply:
It wouldn't be UB in this case, as `ID3v2Tag::insert` and `ID3v2Tag::set_comment` are being used. Those both will remove any collisions prior to insertion, so it would be safe to merge even if the tag has been modified.

Reply:
I have revisited the ID3v2 implementation and eliminated some inconsistencies. Debug assertions indicate situations were undefined behavior occurs when the preconditions of this operation are violated.

There was also a bug: Simply appending the picture frames was wrong, because it could violate the ID3v2 restrictions. Deduplication during the key-to-frame mapping by using `insert()` is required here and not covered by the preconditions.